python/cudf/cudf/_lib/pylibcudf/column.pxd

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_view cimport column_view
+from cudf._lib.cpp.types cimport offset_type, size_type
+
+from .gpumemoryview cimport gpumemoryview
+from .types cimport DataType
+
+
+cdef class Column:
+    cdef:
+        # Core data
+        DataType data_type
+        size_type size
+        gpumemoryview data
+        gpumemoryview mask
+        size_type null_count
+        offset_type offset
+        # TODO: Is there a more efficient container I could use than a list
+        # here? For now defaulting to List[gpumemoryview]
+        object children
+
+        # Internals
+        unique_ptr[column_view] _underlying
+
+    cdef column_view* get_underlying(self)
+
+    @staticmethod
+    cdef Column from_libcudf(unique_ptr[column] libcudf_col)
```

Comment:
Do we want to consistently name these factories `from_libcudf`? In `DataType` for example the equivalent constructor is named `from_data_type`.

Reply:
Good call. I think I went back and forth during development and didn't make them consistent. I will do so now.

Reply:
I renamed the DataType method. Do you see any others? I think that was the only other one.